File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/params/get_cartesian_rotation_max_vel.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w
import time

h10w_params = h10w.H10wParams()
time.sleep(1)

print("=== 获取笛卡尔旋转最大速度参数示例 ===")

ret, vel_list = h10w_params.getCartRotaMaxVel()
if ret == 0:
    print("获取成功")
    for index, vel in vel_list:
        print("    笛卡尔轴 {}: 最大旋转速度= {:.3f} rad/s".format(index, vel))
else:
    print("获取失败")

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/params/get_cartesian_rotation_max_vel.cpp
// get_cartesian_rotation_max_vel.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_params.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取笛卡尔旋转最大速度参数示例 ===" << std::endl;
  H10wParams params;
  CartRotaMaxVel cartRotaMaxVel;

  if (params.getCartRotaMaxVel(cartRotaMaxVel) == 0) {
    std::cout << "获取成功" << std::endl;

    for (const auto& vel : cartRotaMaxVel) {
      const auto& [cartIndex, maxVel] = vel;
      std::cout << "    笛卡尔轴 " << cartIndex << ": "
                << "最大旋转速度=" << std::fixed << std::setprecision(3)
                << maxVel << " rad/s" << std::endl;
    }
  } else {
    std::cout << "获取失败" << std::endl;
    return -1;
  }
  return 0;
}